integration/container/mounts_linux_test.go

```diff
@@ -82,3 +92,63 @@ func TestContainerShmNoLeak(t *testing.T) {
 		t.Fatalf("mount leaked: %s", string(out))
 	}
 }
+
+func TestContainerNetworkMountsNoChown(t *testing.T) {
+	// chown only applies to Linux bind mounted volumes; must be same host to verify
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux" || !testEnv.IsLocalDaemon())
+
+	defer setupTest(t)()
+
+	ctx := context.Background()
+
+	tmpDir, err := ioutils.TempDir("", "test-network-mounts")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	// make tmp dir readable by anyone to allow userns process to mount from
+	err = os.Chmod(tmpDir, 0755)
+	require.NoError(t, err)
+	// create temp files to use as network mounts
+	tmpNWFileMount := filepath.Join(tmpDir, "nwfile")
+
+	err = ioutil.WriteFile(tmpNWFileMount, []byte("network file bind mount"), 0644)
+	require.NoError(t, err)
```

Comment:
not required, but you could do all of this tmpdir setup work with `gotestyourself/fs`:

```go
tmpDir := fs.NewDir(t, "network-file-mounts",
    fs.WithMode(0755),
    fs.WithFile("nwfile", "network file bind mount", fs.WithMode(0644))
defer tmpDir.Remove()
```

Reply:
Thanks @dnephin. The PR has been updated.